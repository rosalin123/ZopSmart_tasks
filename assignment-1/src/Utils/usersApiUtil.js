const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url, { method: 'GET' });
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {

    let url = `${baseUrl}/users/${id}`;
    let user = await fetch(url, { method: 'GET' });
    let userJson = await user.json();
    return userJson;

};
