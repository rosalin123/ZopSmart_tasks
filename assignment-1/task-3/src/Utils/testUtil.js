import React from 'react';

export const findbyTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
