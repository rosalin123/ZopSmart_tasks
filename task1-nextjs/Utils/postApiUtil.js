import axios from 'axios';
import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchPosts = async () => {
  let url = `${baseUrl}/posts`;
  let posts = await axios.get(url);
  let postsJson = await posts.json();
  return postsJson;
};

export const getPost = async (id) => {
  let url = `${baseUrl}/posts/${id}`;
  let post = fetch(url);
  let postJson = await post.json();
  return postJson;
};

export const getPostComments = async (id) => {
  let url = `${baseUrl}/posts/${id}/comments`;
  let comments = await axios.get(url);
  let commentsJson = await comments.json();
  return commentsJson;
};

export const getUserPosts = async (id) => {
  let url = `${baseUrl}/posts?userId=${id}`;
  let userPosts = await fetch(url);
  let userPostsJson = await userPosts.json();
  return userPostsJson;
};
