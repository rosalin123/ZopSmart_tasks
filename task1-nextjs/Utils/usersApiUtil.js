import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url);
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {
  let url = `${baseUrl}/users/${id}`;
  let user = await fetch(url);
  let userJson = await user.json();
  return userJson;
};
