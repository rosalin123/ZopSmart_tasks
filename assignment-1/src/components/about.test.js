import React from 'react';
import { shallow } from 'enzyme';
import About from './about';
import { findbyTestAttr } from '../Utils/testUtil';

let setUp = (props) => {
  let component = shallow(<About {...props} />);
  return component;
};

describe('About Component', () => {
  let component;

  beforeEach(() => {
    let props = {
      name: 'rose',
      username: 'rosy',
      email: 'r@gmail.com',
      company: { name: 'abc' },
      address: { street: 'abc', city: 'def', zipcode: '123', suite: 'ghi' },
      phone: '123',
      website: 'aa.com',
    };
    component = setUp(props);
  });

  it('About should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });

  // it('handleClick method should update state as expected', () => {
  //   const classInstance = component.instance();
  //   classInstance.handleClick('work');
  //   const newState = classInstance.state.work;
  //   expect(newState).toBe(true);
  // });
  // it('should render header', () => {
  //   let wrapper = findbyTestAttr(component, 'header');
  //   expect(wrapper.length).toBe(1);
  // });

  // it('should render buttons', () => {
  //   let wrapper = findbyTestAttr(component, 'buttons');
  //   expect(wrapper.length).toBe(1);
  // });

  // it('should render detailsButton', () => {
  //   let wrapper = findbyTestAttr(component, 'detailsButton');
  //   expect(wrapper.length).toBe(1);
  // });

  // it('should render workDetailsButton', () => {
  //   let wrapper = findbyTestAttr(component, 'buttons');
  //   expect(wrapper.length).toBe(1);
  // });

  // it('should render addressButton', () => {
  //   let wrapper = findbyTestAttr(component, 'addressButton');
  //   expect(wrapper.length).toBe(1);
  // });

  // it('should render contactInfoButton', () => {
  //   let wrapper = findbyTestAttr(component, 'contactInfoButton');
  //   expect(wrapper.length).toBe(1);
  // });

  // it('should render details', () => {
  //   let wrapper = findbyTestAttr(component, 'details');
  //   expect(wrapper.length).tobe(1);
  // });
});
