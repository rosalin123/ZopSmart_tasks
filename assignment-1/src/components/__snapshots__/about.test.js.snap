// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component About should render without throwing an error 1`] = `
<About
  address={
    Object {
      "city": "def",
      "street": "abc",
      "suite": "ghi",
      "zipcode": "123",
    }
  }
  classes={
    Object {
      "aboutStyles": "About-aboutStyles-2",
      "borderStyles": "About-borderStyles-3",
      "buttonStyles": "About-buttonStyles-4",
      "hoverStyles": "About-hoverStyles-5",
      "pos": "About-pos-7",
      "root": "About-root-1",
      "title": "About-title-6",
    }
  }
  company={
    Object {
      "name": "abc",
    }
  }
  email="r@gmail.com"
  name="rose"
  phone="123"
  username="rosy"
  website="aa.com"
/>
`;
